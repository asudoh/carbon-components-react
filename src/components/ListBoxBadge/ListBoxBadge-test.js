import React from 'react';
import { shallow } from 'enzyme';
import ListBoxBadge from '../ListBoxBadge';

describe('ListBoxBadge component', () => {
  let mockClearSelection;

  beforeEach(() => {
    mockClearSelection = jest.fn();
  });

  it('should render', () => {
    const wrapper = shallow(
      <ListBoxBadge clearSelection={mockClearSelection} count={5} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render nothing if the count is 0', () => {
    const wrapper = shallow(
      <ListBoxBadge clearSelection={mockClearSelection} count={0} />
    );
    expect(wrapper.type()).toEqual(null);
  });

  it('should render a non-zero count as badge text', () => {
    const count = 5;
    const wrapper = shallow(
      <ListBoxBadge clearSelection={mockClearSelection} count={count} />
    );
    expect(wrapper.find('.bx--list-box__badge-text').text()).toBe(`${count}`);
  });

  it('should call clearSelection when clicked on', () => {
    const wrapper = shallow(
      <ListBoxBadge clearSelection={mockClearSelection} count={5} />
    );
    wrapper.simulate('click', { stopPropagation: () => {} });
    expect(mockClearSelection).toHaveBeenCalledTimes(1);
  });

  it('should call clearSelection when the enter key is pressed', () => {
    const wrapper = shallow(
      <ListBoxBadge clearSelection={mockClearSelection} count={5} />
    );
    wrapper.simulate('keyDown', { keyCode: 8 });
    expect(mockClearSelection).not.toHaveBeenCalled();
    wrapper.simulate('keyDown', { keyCode: 13 });
    expect(mockClearSelection).toHaveBeenCalledTimes(1);
  });
});
