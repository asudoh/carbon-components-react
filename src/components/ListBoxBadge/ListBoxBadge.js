import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

const ListBoxBadge = ({ clearSelection, count }) => {
  // We shouldn't render anything if we haven't selected anything yet
  if (count === 0) {
    return null;
  }

  const handleOnClick = event => {
    // Stopping the propogation here allows us to not trigger any
    // downshift-related events
    event.stopPropagation();
    clearSelection();
  };
  const handleOnKeyDown = event => {
    // When a user hits ENTER, we'll clear the selection
    if (event.keyCode === 13) {
      clearSelection();
    }
  };

  return (
    <div
      className="bx--list-box__badge"
      title="Clear all selected items"
      tabIndex="0"
      onClick={handleOnClick}
      onKeyDown={handleOnKeyDown}>
      <Icon
        name="close--glyph"
        className="bx--list-box__clear"
        description="Clear all selected items"
      />
      <span className="bx--list-box__badge-text">{count}</span>
    </div>
  );
};

ListBoxBadge.propTypes = {
  count: PropTypes.number.isRequired,
  clearSelection: PropTypes.func.isRequired,
};

export default ListBoxBadge;
