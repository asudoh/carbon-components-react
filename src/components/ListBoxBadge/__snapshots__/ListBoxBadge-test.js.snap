// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxBadge component should render 1`] = `
<div
  className="bx--list-box__badge"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
  title="Clear all selected items"
>
  <Icon
    className="bx--list-box__clear"
    description="Clear all selected items"
    fillRule="evenodd"
    name="close--glyph"
    role="img"
  />
  <span
    className="bx--list-box__badge-text"
  >
    5
  </span>
</div>
`;
